packages/core/src/components/forms/inputGroup.tsx

```diff
@@ -106,17 +99,15 @@ export class InputGroup extends React.Component<HTMLInputProps & IInputGroupProp
         );
     }
 
-    private updateInputWidth() {
-        if (this.rightElement != null) {
-            const { clientWidth } = this.rightElement;
-            // small threshold to prevent infinite loops
-            if (Math.abs(clientWidth - this.state.rightElementWidth) > 2) {
-                this.setState({ rightElementWidth: clientWidth });
-            }
-        } else {
-            this.setState({ rightElementWidth: 0 });
-        }
-    }
+    private handleOnBlur = (e: React.FocusEvent<HTMLInputElement>) => {
+        this.props.onBlur != null ? this.props.onBlur(e) : undefined;
+        this.setState({ isActive: !this.state.isActive });
+    };
+
+    private handleOnFocus = (e: React.FocusEvent<HTMLInputElement>) => {
+        this.props.onFocus != null ? this.props.onFocus(e) : undefined;
```

Comment:
we have a hammer for this: `Utils.safeInvoke(this.props.onFocus, e)`

Reply:
What's a hammer? So it should be `this.props.onFocus != null ? Utils.safeInvoke(this.props.onFocus, e) : undefined;`?

Reply:
no, just safeInvoke. search codebase for other usages, you'll see what it does.

Reply:
basically, `safeInvoke` does this logic for you. line should be `Utils.safeInvoke(this.props.onFocus, e);`

Reply:
@llorca @giladgray stashing this and punting for now